tests/functional/test_service_headers.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import pytest
+
+from botocore.session import get_session
+
+# This test verifies that S3 is the only service that uses the 'expires' header
+# in their response. If any other service is found to target this header in any
+# of their output shapes, the test will fail and alert us of the change.
+
+
+@pytest.mark.validates_models
+def test_only_s3_targets_expires_header():
+    session = get_session()
+    loader = session.get_component('data_loader')
+    services = loader.list_available_services('service-2')
+    services_that_target_expires_header = set()
+
+    for service in services:
+        service_model = session.get_service_model(service)
+        for operation in service_model.operation_names:
+            operation_model = service_model.operation_model(operation)
+            if output_shape := operation_model.output_shape:
+                if _shape_targets_expires_header(output_shape):
+                    services_that_target_expires_header.add(service)
+    assert services_that_target_expires_header == {'s3'}
```

Comment:
Can we make this a more descriptive failure so we can more easily detect the issue?